target_chains/ethereum/contracts/forge-test/Entropy.t.sol

```diff
@@ -729,4 +731,161 @@ contract EntropyTest is Test, EntropyTestUtils {
         vm.expectRevert();
         random.setProviderUri(newUri);
     }
+
+    function testRequestWithCallbackAndReveal() public {
+        bytes32 userRandomNumber = bytes32(uint(42));
+        uint fee = random.getFee(provider1);
+        EntropyStructs.ProviderInfo memory providerInfo = random
+            .getProviderInfo(provider1);
+
+        vm.roll(1234);
+        vm.deal(user1, fee);
+        vm.startPrank(user1);
+        vm.expectEmit(false, false, false, true, address(random));
+        emit RequestedWithCallback(
+            provider1,
+            user1,
+            providerInfo.sequenceNumber,
+            userRandomNumber,
+            EntropyStructs.Request({
+                provider: provider1,
+                sequenceNumber: providerInfo.sequenceNumber,
+                numHashes: SafeCast.toUint32(
+                    providerInfo.sequenceNumber -
+                        providerInfo.currentCommitmentSequenceNumber
+                ),
+                commitment: keccak256(
+                    bytes.concat(
+                        random.constructUserCommitment(userRandomNumber),
+                        providerInfo.currentCommitment
+                    )
+                ),
+                blockNumber: 1234,
+                requester: user1,
+                useBlockhash: false,
+                isRequestWithCallback: true
+            })
+        );
+        vm.roll(1234);
+        uint64 assignedSequenceNumber = random.requestWithCallback{value: fee}(
+            provider1,
+            userRandomNumber
+        );
+
+        assertEq(
+            random.getRequest(provider1, assignedSequenceNumber).requester,
+            user1
+        );
+
+        assertEq(
+            random.getRequest(provider1, assignedSequenceNumber).provider,
+            provider1
+        );
+
+        vm.expectRevert();
+        random.reveal(
+            provider1,
+            assignedSequenceNumber,
+            userRandomNumber,
+            provider1Proofs[assignedSequenceNumber]
+        );
+        vm.stopPrank();
+    }
+
+    function testRequestAndRevealWithCallback() public {
```

Comment:
I think this test should be called `testRequestWithCallbackAndRevealWithCallback`

Reply:
There is no test for RequestWIthCallbackAndReveal(without Callback)